lib/mongoid/contextual/mongo.rb

```diff
@@ -703,21 +792,13 @@ def acknowledged_write?
       #
       # @param [ Object ] The value for the given field name
       def extract_value(attrs, field_name)
-        def fetch_and_demongoize(d, meth, klass)
-          res = d.try(:fetch, meth, nil)
-          if field = klass.fields[meth]
-            field.demongoize(res)
-          else
-            res.class.demongoize(res)
-          end
-        end
-
         i = 1
         num_meths = field_name.count('.') + 1
         k = klass
```

Comment:
@Neilshweky it seems that as a result of this PR, this local variable `k = klass` is no longer used (it is assigned also on line 837, but it's never read.)